Import latest changes from nne_dev branch

joshih-cad opened the pull request:
1. 32b bias handling for 8x16 FC and Conv in CREF and Xtensa code.
2. Unpack INT16 support

joshih-cad committed:
Import latest changes from nne_dev branch
1. 32b bias handling for 8x16 FC and Conv in CREF and Xtensa code.
2. Unpack INT16 support

cad-audio commented on tensorflow/lite/micro/kernels/conv.cc:
```diff
@@ -101,7 +101,7 @@ TfLiteStatus ConvEval(TfLiteContext* context, TfLiteNode* node) {
 #endif  // USE_TFLM_COMPRESSION
```
To be consistent, this should be GetTensorData(), right?

joshih-cad replied:
PR updated, marking as resolved

cad-audio commented on tensorflow/lite/micro/kernels/conv.cc:
```diff
@@ -118,7 +118,7 @@ TfLiteStatus ConvEval(TfLiteContext* context, TfLiteNode* node) {
 #else   // USE_TFLM_COMPRESSION
```
This should be GetOptionalTensorData() ?

joshih-cad replied:
Yes, will update the PR

cad-audio commented on tensorflow/lite/micro/kernels/fully_connected.cc:
```diff
@@ -253,7 +253,7 @@ TfLiteStatus FullyConnectedEval(TfLiteContext* context, TfLiteNode* node) {
+                      tflite::micro::GetTensorData<int32_t>(
                           micro_context, bias, bias_comp_td,
                           data.bias_scratch_index),
 #else   // USE_TFLM_COMPRESSION
```
Should be GetTensorData()?

joshih-cad committed:
Handle bias=null for FC,Conv 8x16 CREF code
- Streamlined the usage of GetTensorData() and GetOptionaltensorData()
- Added bias != nullptr condiition check wherever necessary.

joshih-cad committed:
Missed file during prev commit
